src/main/kotlin/com/openlattice/entitysets/controllers/EntitySetsController.kt

```diff
@@ -402,6 +414,87 @@ constructor(
         return 1
     }
 
+    @Timed
+    @RequestMapping(
+            path = [ALL + ID_PATH + EXPIRATION_PATH],
+            method = [RequestMethod.PATCH],
+            produces = [MediaType.APPLICATION_JSON_VALUE]
+    )
+    override fun removeDataExpirationPolicy(
+            @PathVariable(ID) entitySetId: UUID
+    ): Int {
+        ensureOwnerAccess(AclKey(entitySetId))
+        edmManager.removeDataExpirationPolicy(entitySetId)
+
+        recordEvent(
+                AuditableEvent(
+                        getCurrentUserId(),
+                        AclKey(entitySetId),
+                        AuditEventType.UPDATE_ENTITY_SET,
+                        "Entity set data expiration policy removed through EntitySetApi.removeDataExpirationPolicy",
+                        Optional.empty(),
+                        ImmutableMap.of("entitySetId", entitySetId),
+                        OffsetDateTime.now(),
+                        Optional.empty()
+                )
+        )
+        return 1
+    }
+
+    @Timed
+    @RequestMapping(
+            path = [ALL + ID_PATH + EXPIRATION_PATH],
+            method = [RequestMethod.POST]
+    )
+    override fun getExpiringEntitiesFromEntitySet(
+            @PathVariable(ID) entitySetId: UUID,
+            @RequestBody dateTimeAsString: String): Set<UUID> {
+        ensureReadAccess(AclKey(entitySetId))
```

Comment:
I'm not sure this needs to be exposed via the API, since I think this will only be called internally when removing expired data, and this is a lot of work for datastore to be doing. This seems like it should go in a DataExpirationService or something in conductor-client or indexer

Reply:
^nvm ignore this, I realize this is used for loading soon-to-expire data.

 I would move `getSqlParameters` out of the controller and into the service though

Reply:
The ticket asked for the ability to review "soon to expire data", is there a way to allow this without an api?

Reply:
Ah okay will do